third-culture-software/bhima
# Feature: move delete records to a role_action instead of an enterprise_setting

This commit moves the delete records value to a role_action instead of an enterprise_setting.  This change makes it so that the deletion ability is partitioned by user instead of just being globally "on" or "off" for an entire enterprise.

Further, we've also separated out each type of deletion so that an enterprise can choose to allow the deletion of only a certain type of transaction (for example, purchase orders) without allowing any others.  Alternatively, they can give some users some permissions but not all users all permissions.

Here is what it looks like:

![rod4ARuPsJ](https://user-images.githubusercontent.com/896472/114709143-23cdb780-9d24-11eb-990a-72856d7a2ba2.gif)



How to tests this:
  1. Build the test database.
  2. Sign in as the superuser
  3. Go to the invoice registry
  4. Observe that you have access to delete records in the dropdown menu.
  5. Navigate to the posting journal.
  6. Observe that you have access to the "unpost" button at the top of the journal.
  7. Navigate to the roles module. 
  8. Edit the superuser **actions** and uncheck the "Can unpost transactions" option and the "Delete Invoice" options.
  9. Submit.
  9. Navigate to both modules and observe that you no longer have access.
  10. Navigate back and check the options again
  11. You will now have access.

Closes  #5379.
